src/register_hall_of_fame.c

```diff
@@ -0,0 +1,3069 @@
+            9,
+            TRUE,
+            NNS_G2D_VRAM_TYPE_2DMAIN,
+            55518
+        );
+        SpriteRenderer_LoadCellResObjFromOpenNarc(
+            data->spriteRenderer,
+            data->spriteGfxHandler,
+            data->narcA094,
+            10,
+            TRUE,
+            55514
+        );
+        SpriteRenderer_LoadAnimResObjFromOpenNarc(
+            data->spriteRenderer,
+            data->spriteGfxHandler,
+            data->narcA094,
+            11,
+            TRUE,
+            55514
+        );
+    }
+    SpriteRenderer_LoadPlttResObjFromOpenNarc(
+        data->spriteRenderer,
+        data->spriteGfxHandler,
+        data->narcA094,
+        15,
+        FALSE,
+        1,
+        NNS_G2D_VRAM_TYPE_2DMAIN,
+        55518
+    );
+    SpriteRenderer_LoadPlttResObjFromOpenNarc(
+        data->spriteRenderer,
+        data->spriteGfxHandler,
+        data->narcA094,
+        15,
+        FALSE,
+        1,
+        NNS_G2D_VRAM_TYPE_2DMAIN,
+        55519
+    );
+}
+
+static void RegisterHallOfFame_IndivMonsScene_UnloadMonOverworldSprite(RegisterHallOfFameData *data) {
+    SpriteGfxHandler_UnloadPlttObjById(data->spriteGfxHandler, 55518);
+    SpriteGfxHandler_UnloadPlttObjById(data->spriteGfxHandler, 55519);
+    SpriteGfxHandler_UnloadCharObjById(data->spriteGfxHandler, 55518);
+    SpriteGfxHandler_UnloadCellObjById(data->spriteGfxHandler, 55514);
+    SpriteGfxHandler_UnloadAnimObjById(data->spriteGfxHandler, 55514);
+}
+
+static void RegisterHallOfFame_IndivMonsScene_SetMon3dSpriteTex(RegisterHallOfFameData *data, RegisterHofMon *hofMon, int picIdx) {
+    NNSG3dResTex *resTex;
+    void *fileData;
+    const void *sp18;
+    u32 sp14 = NNS_G2dGetImageLocation(Sprite_GetImageProxy(data->monPics[picIdx]->sprite), NNS_G2D_VRAM_TYPE_2DMAIN);
+    int sp10 = hofMon->tsure_param[1] ? 8 : 4;
+    u32 size = 32 * sp10 * sp10;
+    int fileno = RegisterHallOfFame_GetMmodelBySpeciesFormGender(hofMon->species, hofMon->form, hofMon->gender);
+    fileData = AllocAndReadWholeNarcMemberByIdPair(NARC_data_mmodel_mmodel, fileno, HEAP_ID_REGISTER_HALL_OF_FAME);
+    resTex = NNS_G3dGetTex(fileData);
+    sp18 = NNS_G3dGetTexData(resTex);
+    void *buffer = AllocFromHeap(HEAP_ID_REGISTER_HALL_OF_FAME, size);
+    for (u8 i = 0; i < 8; ++i) {
+        sub_020145B4((const u8 *)sp18 + size * i, sp10, 0, 0, sp10, sp10, buffer);
+        RegisterHallOfFame_ReplaceSpriteChar(buffer, sp14 + size * i, size);
+    }
+    FreeToHeap(buffer);
+
+    u32 plttLoc = NNS_G2dGetImagePaletteLocation(Sprite_GetPaletteProxy(data->monPics[picIdx]->sprite), NNS_G2D_VRAM_TYPE_2DMAIN);
+    const void *loadPos = NNS_G3dGetPlttData(resTex);
+    if (MonIsShiny(hofMon->mon) == TRUE) {
+        loadPos = (const u8 *)loadPos + 0x20;
+    }
+    RegisterHallOfFame_ReplaceSpritePltt(loadPos, plttLoc, 0x20);
+    FreeToHeap(fileData);
+}
+
+static void RegisterHallOfFame_IndivMons_InitBgLayers(RegisterHallOfFameData *data) {
+    {
```

Comment:
unnecessary(?) scope brackets

Reply:
this is the style i'm using for all routines following this pattern of "set graphics modes then init several bg layers".